reference/features/security.md

```diff
@@ -0,0 +1,237 @@
+# Security
+
+MeiliSearch gives you fine-grained control over which users can access which indexes, endpoints, and routes. When protecting your instance with a master key, you can ensure only authorized users can carry out sensitive tasks such as adding documents or altering index settings.
+
+In this guide, you will learn how to protect your MeiliSearch instance with a master key and communicate with it using an API key. You will also see how you can use your instance's master key to create, update, list and delete API keys.
+
+## Protecting a MeiliSearch instance
+
+By default, MeiliSearch is unprotected. This means all routes are publicly accessible.
+
+To protect a MeiliSearch instance from unauthorized use, you must supply an alphanumeric string at launch. This alphanumeric string is your master key.
+
+Setting up a master key can be done with either command-line options or environment variables. You can read more about [master key configuration in our instance configuration guide](/reference/features/configuration.md#master-key).
+
+:::: tabs
+::: tab CLI
+
+```sh
+./meilisearch --master-key="[your_master_key]"
+```
+
+:::
+
+::: tab Environment variable
+
+```sh
+MEILI_MASTER_KEY="[your_master_key]"
+./meilisearch
+```
+
+:::
+::::
+
+::: note
+Launching an instance with a master key is mandatory when `env` is set to `production`.
+:::
+
+Once you launch MeiliSearch with a master key, the instance is protected. From that point on, [communicating with a protected instance](#communicating-with-a-protected-instance) requires adding an `Authorization` header to requests.
+
+## Communicating with a protected instance
+
+After an instance is secured, only the [`GET /health` endpoint](/reference/api/health.md) will be publicly available. To access any other API endpoint, you must add a security key with suitable permissions to your request.
```

Comment:
I wanted to send this back to review today, so I took a different approach.

I incorporated your edits to the first paragraph, but instead of creating a new subheading here explaining what API keys are, I left the code sample still close to the heading. Then, I moved a renamed "Default API keys" section just after the example and opened a new section named "Differences between the master key and API keys".

It's likely not the most elegant or ideal solution, but it was quicker and, I hope, address the main issues with content labeling and information architecture.